tests/IResearch/CMakeLists.txt

```diff
@@ -1,118 +1,111 @@
+  IResearchLinkCoordinatorTest.cpp
+  IResearchLinkHelperTest.cpp
+  IResearchLinkMetaTest.cpp
+  IResearchOrderTest.cpp
+  IResearchQueryAggregateTest.cpp
+  IResearchQueryAndTest.cpp
+  IResearchQueryBooleanTermTest.cpp
+  IResearchQueryComplexBooleanTest.cpp
+  IResearchQueryExistsTest.cpp
+  IResearchQueryFilterTest.cpp
+  IResearchQueryGeoIntersectsTest.cpp
+  IResearchQueryGeoContainsTest.cpp
+  IResearchQueryGeoDistanceTest.cpp
+  IResearchQueryGeoInRangeTest.cpp
+  IResearchQueryInTest.cpp
+  IResearchQueryInRangeTest.cpp
+  IResearchQueryJoinTest.cpp
+  IResearchQueryLateMaterializationTest.cpp
+  IResearchQueryMinMatchTest.cpp
+  IResearchQueryNGramMatchTest.cpp
+  IResearchQueryNoMaterializationTest.cpp
+  IResearchQueryNullTermTest.cpp
+  IResearchQueryNumericTermTest.cpp
+  IResearchQueryOptimizationTest.cpp
+  IResearchQueryOptionsTest.cpp
+  IResearchQueryOrTest.cpp
+  IResearchQueryPhraseTest.cpp
+  IResearchQueryLevenshteinMatchTest.cpp
+  IResearchQueryWildcardTest.cpp
+  IResearchQueryScorerTest.cpp
+  IResearchQuerySelectAllTest.cpp
+  IResearchQueryStartsWithTest.cpp
+  IResearchQueryStringTermTest.cpp
+  IResearchQueryTokensTest.cpp
+  IResearchQueryValueTest.cpp
+  IResearchQueryTraversalTest.cpp
+  IResearchPrimaryKeyReuseTest.cpp
+  IResearchTestCommon.cpp
+  IResearchViewTest.cpp
+  IResearchViewCoordinatorTest.cpp
+  IResearchViewDBServerTest.cpp
+  IResearchViewMetaTest.cpp
+  IResearchViewNodeTest.cpp
+  IResearchViewSortTest.cpp
+  IResearchViewSortedTest.cpp
+  IResearchViewCountApproximateTest.cpp
+  VelocyPackHelperTest.cpp)
 
-set(ARANGODB_IRESEARCH_TESTS_SOURCES
-  ${ARANGODB_IRESEARCH_TESTS_SOURCES}
-  PARENT_SCOPE)
+if (USE_JEMALLOC)
+  add_dependencies(arango_tests_iresearch jemalloc_build)
+endif ()
 
-add_executable(
-  arangodbtests-iresearch
-  ${ARANGODB_IRESEARCH_TESTS_SOURCES}
-)
+target_link_libraries(arango_tests_iresearch
+  arango_tests_basics
+  arango
+  arango_agency
+  arango_cluster_engine
+  arango_rocksdb
+  arango_v8server
+  arangoserver
+  gtest
+  boost_boost
+  v8_interface)
 
-set_target_properties(arangodbtests-iresearch PROPERTIES EXCLUDE_FROM_ALL TRUE)
+add_executable(arangodbtests_iresearch
+  EXCLUDE_FROM_ALL)
 
 #required by fakeit dtor mock -- https://github.com/eranpeer/FakeIt/issues/84
-if(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
-  target_compile_options(arangodbtests-iresearch PRIVATE -fno-devirtualize)
-endif()
+if (CMAKE_CXX_COMPILER_ID STREQUAL GNU)
+  target_compile_options(arangodbtests_iresearch PRIVATE -fno-devirtualize)
+endif ()
 
-target_link_libraries(arangodbtests-iresearch
-  gtest
+target_link_libraries(arangodbtests_iresearch
   arango_tests_basics
+  arango
   arango_agency
   arango_cluster_engine
   arango_rocksdb
   arango_v8server
   arangoserver
   boost_boost
-)
-
-target_include_directories(arangodbtests-iresearch PRIVATE
-  ${PROJECT_SOURCE_DIR}/arangod
-  ${PROJECT_SOURCE_DIR}/${ENTERPRISE_INCLUDE_DIR}
-  ${PROJECT_SOURCE_DIR}/tests
-)
+  v8_interface
```

Comment:
I don't think that you have to link against all these libraries again after you linked them already into arango_tests_iresearch.

Reply:
I think you're right but It's copied from graph code https://github.com/arangodb/arangodb/blob/devel/tests/Graph/CMakeLists.txt#L30

Reply:
About fake-it, for me it works without it.

And also as I remeber fake-it is part of arango_tests_basics

target_include_directories(arango_tests_iresearch PUBLIC ${PROJECT_SOURCE_DIR}/3rdParty/fakeit-gtest/)

Reply:
You're right, I've improved the graph cmake in https://github.com/arangodb/arangodb/pull/17941.

It's fine for me to not use fake it, but currently your code does not compile for me with error: `https://github.com/arangodb/arangodb/pull/17941`